src/app/common/utils/helpers/inheritance-resolver.js

```diff
@@ -41,7 +41,15 @@ export default class InheritanceResolver {
 
     if (referenceDataStructure === member.element) return !member.content;
 
-    return member.content ? refDSContent.length === member.content.length : false;
+    return member.content
+      ? refDSContent.length === member.content.length
+        && !member.content.some(mc => {
+          const refDSContentItem = refDSContent.find(dsc => dsc.content.key.content === mc.content.key.content);
+
+          return !refDSContentItem
+            || (!mc.usedStructuresHash && JSON.stringify(refDSContentItem.content.value) !== JSON.stringify(mc.content.value));
```

Comment:
I understand intentions behind these changes, but I think it may be too heavy to process in runtime.
Now we have here loop inside of loop (some → find) which is already n^2 complexity, and in every step also has double json.stringify call.

I am not sure that it is so necessary to do. I would try to stick to the second option where you cache only structures without nested members.

Reply:
done